GravityKit/GravityView
# Select different layouts for multiple and single entry (per-context switcher)

## Things to test:

- [x] Saving existing Views shouldn't change anything
  - Edit an existing View
  - Don't do anything
  - Save the View
  - Check to make sure the previous single/multiple entries layouts don't change
- [x] Updating a View works
  - Edit an existing View
  - Change the layout type
  - Save the View
  - Confirm that the layouts are correct for single & multiple layouts
- [x] Confirm that the View type change dialog occurs when switching from Table/DataTables to other layouts
- [x] Confirm that when changing layout types and the fields are cleared, only the current tab is affected

### Testing different states
- Unlicensed or license doesn't support layouts
  - [x] Should show "Available in Pro" button
  - [x] Items should [look like the design](https://www.figma.com/file/OM63Lq6eIlzu3nFPR4Xro8/GravityKit-UI%2FUX?type=design&node-id=2851-1934&mode=design&t=kHZnBwMUU83JZgEr-0)
- Has license but layout is not installed
  - Should [look like design](https://www.figma.com/file/OM63Lq6eIlzu3nFPR4Xro8/GravityKit-UI%2FUX?type=design&node-id=2851%3A107&mode=design&t=kHZnBwMUU83JZgEr-1)
  - [x] Clicking the Install Now button should install and activate the layout plugin
  - [x] The layout should then appear clickable
  - [x] Clicking the newly-installed layout should activate it
- Has license and layout is installed, but not activated
  - [x] Clicking Activate should activate the layout plugin
  - [x] The layout should then appear clickable
  - [x] Clicking the newly-installed layout should activate it

### Browser test

Check appearance, confirm functionality in:


-------

This PR addresses #1177. 

It adds a new dropdown solution with support for icons and descriptions. It also adds a new variable that contains a separate template for the single entry. There is backwards compatibility as the value of the new variable has a fallback to the "old" variable, that is now solely for the multiple entries.

![pr-preview](https://github.com/GravityKit/GravityView/assets/529515/7bf0c692-1a42-47b1-ac2e-1ad22d3d4033)


Todo: 
- [x] It needs a design (and fixes) for inactive items like the regular selection now has.
- [x] It needs icons for native templates, otherwise fallback to the image it now has.

In the meantime this can be tested and reviewed for what we have now. 

![image](https://github.com/GravityKit/GravityView/assets/529515/b62f25e5-a96c-4bbc-bda5-41bcd53c1009)

💾 [Build file](https://www.dropbox.com/scl/fi/zkz7kc5lu38n3f2m3bqj3/gravityview-2.23-18613244e.zip?rlkey=2i9n51967fewbpdeagd15n6wa&dl=1) (18613244e).
